youtube_dl/extractor/megatvcom.py

```diff
@@ -0,0 +1,186 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import hashlib
+import re
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_str,
+    compat_parse_qs,
+    compat_urllib_parse_urlparse,
+)
+from ..utils import (
+    HEADRequest,
+    ExtractorError,
+    determine_ext,
+    get_element_by_class,
+    unified_timestamp,
+    extract_attributes,
+    clean_html,
+    unescapeHTML,
+)
+
+
+class MegaTVComBaseIE(InfoExtractor):
+    _PLAYER_DIV_ID = 'player_div_id'
+
+    def _extract_player_attrs(self, webpage):
+        PLAYER_DIV_RE = r'''(?x)
+        <div(?:
+            id=(?P<_q1>[\"\'])(?P<%(pdi)s>%(pdi)s)(?P=_q1)|
```

Comment:
Since you're using `'''`, you don't need to escape any quotes (in fact this might break it) unless it's the same one in the start or end of the string, touching the same quote.

Same for the other such strings later in the code.

Reply:
You are right that there is no need to escape quotes, so I will remove them. However I do not think there can be any breakage, at least not in this case, since the backslash makes no difference as to what the pattern matches:
```py3
patterns = (r'''["']''', r'''[\"']''', r'''["\']''', r'''[\"\']''',)
strings = ('"', '\"', "'", "\'")
ctx = 'div id=%swhatever'
all(re.match(ctx % p, ctx % s) for p in patterns for s in strings) # True
```

I noticed there are other such cases in the code base (there are more, these are on the same line):
```
youtube_dl/extractor/generic.py:3119:            r'''(?x)<iframe[^>]+src=(["\'])(?P<url>(?:https?:)?//
youtube_dl/extractor/myspace.py:113:                r'''<button.*data-song-id=(["\'])%s\1.*''' % video_id,
youtube_dl/extractor/peertube.py:498:            r'''(?x)<iframe[^>]+\bsrc=["\'](?P<url>(?:https?:)?//%s/videos/embed/%s)'''
youtube_dl/extractor/simplecast.py:103:            r'''(?x)<iframe[^>]+src=["\']
```